Framework/Crystal/src/FilterPeaks.cpp

```diff
@@ -84,6 +91,12 @@ void FilterPeaks::exec() {
     filterFunction = &HKL2;
```

Comment:
`switch` does not work with `std::string`

Reply:
My bad, still on a bit a Java hangover

Reply:
\*old man voice\* Back in my day (pre Java 7) not even Java could switch on strings.

Reply:
Although... it could be refectored as such:
```c++
#include <vector>
#include <tuple>
#include <iostream>
#include <unordered_map>
#include <string>

enum class Operator {
    LessThan,
    GreaterThan
};


int main() {
    const std::unordered_map<std::string, Operator> map = 
    {
        {"<", Operator::LessThan},
        {">", Operator::GreaterThan}
    };

    std::string type = "<";
    map.at(type);
}
```

Reply:
Could we not do something with a template functoin and one of the STL comparison function objects: http://en.cppreference.com/w/cpp/utility/functional#Comparisons.

If there was as single template function that accepted a comparator and we did a single `if/else` check outside of the loop to assign a to a function like with the `filterFunction` we could avoid the branching each loop.

Reply:
@samueljackson92 Do you think it is worth making this change?

Reply:
@martyngigg I think so yes.